chevah/compat/tests/normal/test_filesystem.py

```diff
@@ -1356,6 +1376,114 @@ def test_openFileForWriting_truncate(self):
         file_content = mk.fs.getFileContent(self.test_segments)
         self.assertEqual(new_content, file_content)
 
+    def test_openFileForUpdating_non_existing(self):
+        """
+        An error is raised when trying to open a file which does not
+        exists for updating..
+        """
+        path, segments = mk.fs.makePathInTemp()
+
+        with self.assertRaises(OSError) as context:
+            self.filesystem.openFileForUpdating(segments, utf8=False)
+
+        self.assertEqual(errno.ENOENT, context.exception.errno)
+        self.assertEqual(path.encode('utf-8'), context.exception.filename)
+
+        with self.assertRaises(OSError) as context:
+            self.filesystem.openFileForUpdating(segments, utf8=True)
+
+        self.assertEqual(errno.ENOENT, context.exception.errno)
+        self.assertEqual(path.encode('utf-8'), context.exception.filename)
+
+    def test_openFileForUpdating_existing_binary(self):
+        """
+        When a file is opened for updating the previous content is kept and
+        it will allow writing at arbitrary offsets.
+        """
+        content = b'some existing content'
+        new_content = b'more here'
+        # Create initial content.
+        self.test_segments = mk.fs.createFileInTemp(content=content)
+
+        # Write new content into file.
+        test_file = self.filesystem.openFileForUpdating(
```

Comment:
I see we have a test and it works as advertised on our part, now this is confusing. I will do some tests on my machine as well.